docs/framework.md

```diff
@@ -143,3 +144,22 @@ class MyApplication {
 ```
 
 If you are using League\Container (recommended), then you may simply add and share your own classes to the same container.  If you are using some other DI container, then you should use [delegate lookup](https://github.com/container-interop/fig-standards/blob/master/proposed/container.md#14-additional-feature-delegate-lookup) to combine them.
+
+## Using a Custom Configuration Loader
+
+Robo provides a very simple configuration loader. If you wish to use more capable loader, you may opt to do so. Replace the call to `Robo::createConfiguration()` with code similar to the following:
+```
+use Robo\Config\Config;
+use Robo\Config\YamlConfigLoader;
+use Robo\Config\ConfigProcessor;
+
+$config = new Config();
+$loader = new YamlConfigLoader();
+$processor = new ConfigProcessor();
+$processor->extend($loader->load('defaults.yml'));
```

Comment:
Regarding `$processor->add($config->export(), 'default');`, it's fine for BLT to continue to do this for historic reasons, but in the general case, labeling the config export as `'default'` isn't necessarily good advice, because `$config` might contain something other than default values. The "source name" for loaders is to support a Terminus feature where the Config object keeps track of where each configuration item came from. I'm not sure if I should have put those methods into the base implementation or not, but it seemed generally useful to have in the loaders, anyway. Perhaps I should remove it from the Config object, though, and let Terminus' subclasses handle that. In any event, I think I'll add a note to the documentation after the example to put in a `$processor->add($config->export());` if you're extending an existing configuration.